b1ddi/resource_dns_forward_zone.go

```diff
@@ -0,0 +1,395 @@
+}
+
+func resourceConfigForwardZoneRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*b1ddiclient.Client)
+
+	var diags diag.Diagnostics
+
+	resp, err := c.DNSConfigurationAPI.ForwardZone.ForwardZoneRead(
+		&forward_zone.ForwardZoneReadParams{ID: d.Id(), Context: ctx},
+		nil,
+	)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set("comment", resp.Payload.Result.Comment)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("created_at", resp.Payload.Result.CreatedAt.String())
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("disabled", resp.Payload.Result.Disabled)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	externalForwarders := make([]map[string]interface{}, 0, len(resp.Payload.Result.ExternalForwarders))
+	for _, ep := range resp.Payload.Result.ExternalForwarders {
+		externalForwarders = append(externalForwarders, flattenConfigForwarder(ep))
+	}
+	err = d.Set("forward_only", resp.Payload.Result.ForwardOnly)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("fqdn", resp.Payload.Result.Fqdn)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("hosts", resp.Payload.Result.Hosts)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("internal_forwarders", resp.Payload.Result.InternalForwarders)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("mapped_subnet", resp.Payload.Result.MappedSubnet)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("mapping", resp.Payload.Result.Mapping)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("nsgs", resp.Payload.Result.Nsgs)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("parent", resp.Payload.Result.Parent)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("protocol_fqdn", resp.Payload.Result.ProtocolFqdn)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("tags", resp.Payload.Result.Tags)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("updated_at", resp.Payload.Result.UpdatedAt.String())
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	err = d.Set("view", resp.Payload.Result.View)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	}
+	return diags
+}
+
+func resourceConfigForwardZoneUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*b1ddiclient.Client)
+
+	if d.HasChange("fqdn") {
+		d.Partial(true)
+		return diag.FromErr(fmt.Errorf("changing the value of 'fqdn' field is not allowed"))
+	}
+
+	if d.HasChange("view") {
+		d.Partial(true)
+		return diag.FromErr(fmt.Errorf("changing the value of 'view' field is not allowed"))
+	}
```

Comment:
Why only checking these two fields and not the remaining ones?

Reply:
As only these two fields cannot be altered for a forward zone.